### PR TITLE
Added amqp output

### PR DESCRIPTION
**Pull request by ekini**
Please, review my amqp output plugin.

### CODE REVIEW

File: outputs/amqp/amqp_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	"github.com/influxdb/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConnectAndWrite(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+
+	var url = "amqp://" + testutil.GetLocalHost() + ":5672/"
```
Comment (sparrc) [inline outputs/amqp/amqp_test.go]: It looks like you are following the form of the kafka output plugin here. In the case of kafka, we've added a kafka broker to the `scripts/docker-compose.yml` file that the test is connecting to in long mode. I'm not very familiar with AMQP, but is there a similar machine that you could add to the dockerfile? Would a rabbitmq docker container work? I think something like this should work:
```
rabbitmq:
    image: rabbitmq:3-management
    hostname: docker_rabbit
    ports:
        - "15672:15672"
        - "5672:5672"
```

File: outputs/amqp/amqp.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	"github.com/streadway/amqp"
+)
+
+type AMQP struct {
+	// AMQP brokers to send metrics to
+	URL string
+	// AMQP exchange
+	Exchange string
+	// Routing key
+	RoutingKey string
```
Comment (sparrc) [inline outputs/amqp/amqp.go]: should `routingkey` also be in the sampleConfig?

File: outputs/amqp/amqp.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+func (q *AMQP) Write(bp client.BatchPoints) error {
+	if len(bp.Points) == 0 {
+		return nil
+	}
+
+	for _, p := range bp.Points {
+		// Combine tags from Point and BatchPoints and grab the resulting
+		// line-protocol output string to write to Kafka
```
Comment (sparrc) [inline outputs/amqp/amqp.go]: s/Kafka/amqp/g :-)

**Comment by sparrc** [comment]
Could you write a short README for this? See #177

**Comment by sparrc** [comment]
This looks great! Thank you very much! Please just address the comments that I've left and I can merge it :+1:

**Comment by ekini** [comment]
Thanks for feedback, Cameron! I've corrected typos and added README, but I have one conflict now.
I can do a rebase, but I'm not sure what will happen on github after that.

**Comment by sparrc** [comment]
@ekini thanks much for your contribution! I made some tweaks to the documentation and changed the default RoutingTag to "host" instead of "dc", because I thought it would be better to use a tag that is defined by default.

I liked that idea so I also added that same functionality to the Kafka output, since that was just hardcoded to use "host" before.

**Comment by pdyba** [comment]
Is it possible to use SSL/Certificates with rabbit.output ?
I did not find appropriate information in docs and in code it self (I am a python developer so I could be wrong :) ) ? 
Is there a chance that it will be added in next version ?

**Comment by ekini** [comment]
Yes, I'm going to add it in a week.

**Comment by ekini** [comment]
Please, see #536

**Pull request closed without merging**